File: back_end/internal/entity/task.go
package entity

import (
	"regexp"
	"time"

	"github.com/asaskevich/govalidator"
	"github.com/google/uuid"
)

type Task struct {
	Base        `json:"base" valid:"required"`
	Title       string `json:"title" valid:"alphanum,required" gorm:"unique"`
	Description string `json:"description" valid:"-"`
	Status      bool   `json:"status" valid:"-"`
	ListID      string `json:"list_id" valid:"alphanum,notnull"`
	TimeSelect  string `json:"time_select" valid:"-"`
}

func init() {
	govalidator.SetFieldsRequiredByDefault(true)
}

func NewTask(title string, description string, listId string, timeSelect string) (*Task, error) {

	task := &Task{
		// Base:        Base{},
		Title:       title,
		Description: description,
		Status:      false,
		ListID:      listId,
		TimeSelect:  timeSelect,
	}

	err := task.Prepare()

	if err != nil {
		return nil, err
	}
	return task, nil
	// return &Task{}, fmt.Errorf("Invalid Task")
}

func (task *Task) Prepare() error {
	timeNow := time.Now()
	task.ID = uuid.New().String()
	task.CreatedAt = timeNow.Local().String()
	task.UpdatedAt = timeNow.Local().String()
	task.DeletedAt = timeNow.Local().String()
	task.IsDeleted = false

	err := task.validate()
	if err != nil {
		return err
	}
	return nil

}

func (task *Task) validate() error {
	_, err := govalidator.ValidateStruct(task)
	if err != nil {
		return err
	}
	return nil
}

func IsRegex(str string) string {
	regex := regexp.MustCompile(`[a-zA-Zà-úÀ-Ú0-9]`)

	if regex.MatchString(str) {
		return str
	}
	return "caracters requireds"
}

File: front_end/@core/domain/entities/task.ts
export type TaskProps ={
    _id:string;
    _title:string;
    _description?:string;
    _status?:boolean;
    _priority?:boolean;
    _listId?:string;
    _createdAt?:string;
    _updatedAt?:string;
    _deletedAt?:string;
    _isDeleted?:boolean;
    _timeSelect?:string;
}

export class Task {
    constructor(public props: TaskProps) {}

    get _id() {
        return this.props._id;
    }

    get _title() {
        return this.props._title;
    }

    get _description() {
        return this.props._description;
    }
    get _status(){
        return this.props._status;
    }
    get _priority(){
        return this.props._priority;
    }
    get _listId(){
        return this.props._listId;
    }
    get _createdAt(){
        return this.props._createdAt;
    }
    get _updatedAt(){
        return this.props._updatedAt;
    }
    get _deletedAt(){
        return this.props._deletedAt;
    }
    get _isDeleted(){
        return this.props._isDeleted;
    }
    get _timeSelect(){
        return this.props._timeSelect;
    }
    toJSON() {
        return {
            id: this._id,
            title: this._title,
            description: this._description,
            status: this._status,
            priority: this._priority,
            listId: this._listId,
            createdAt: this._createdAt,
            updatedAt: this._updatedAt,
            deletedAt: this._deletedAt,
            isDeleted: this._isDeleted,
            timeSelect: this._timeSelect,
        };
    }
}
